router/sources/target_address_tests.move

```diff
@@ -0,0 +1,248 @@
+        {
+            let target_address= router::get_v2_target_addr(domain_name, subdomain_name_opt);
+            assert!(*option::borrow(&target_address) == user2_addr, 4);
+        };
+
+        // Clear domain target address
+        router::clear_target_addr(user, domain_name, option::none());
+        {
+            let target_address = router::get_v2_target_addr(domain_name, option::none());
+            assert!(option::is_none(&target_address), 5);
+        };
+
+        // Clear domain target address
+        router::clear_target_addr(user, domain_name, subdomain_name_opt);
+        {
+            let target_address = router::get_v2_target_addr(domain_name, subdomain_name_opt);
+            assert!(option::is_none(&target_address), 6);
+        };
+
+    }
+
+    #[test(
+        router = @router,
+        aptos_names = @aptos_names,
+        aptos_names_v2 = @aptos_names_v2,
+        user1 = @0x077,
+        user2 = @0x266f,
+        aptos = @0x1,
+        foundation = @0xf01d
+    )]
+    #[expected_failure(abort_code = 327689, location = aptos_names_v2::domains)]
+    fun test_set_target_address_should_trigger_auto_migration(
+        router: &signer,
+        aptos_names: &signer,
+        aptos_names_v2: &signer,
+        user1: signer,
+        user2: signer,
+        aptos: signer,
+        foundation: signer
+    ) {
+        router::init_module_for_test(router);
+        let users = test_helper::e2e_test_setup(aptos_names, aptos_names_v2, user1, &aptos, user2, &foundation);
+        let user = vector::borrow(&users, 0);
+        let user2 = vector::borrow(&users, 1);
+        let user2_addr = address_of(user2);
+        let domain_name = utf8(b"test");
+        let subdomain_name = utf8(b"test");
+        let subdomain_name_opt = option::some(subdomain_name);
+
+        // Register with v1
+        router::register_domain(user, domain_name, SECONDS_PER_YEAR, option::none(), option::none());
+        router::register_subdomain(
+            user,
+            domain_name,
+            subdomain_name,
+            SECONDS_PER_YEAR,
+            0,
+            false,
+            option::none(),
+            option::none(),
+        );
+
+        // Bump mode
+        router::set_mode(router, 1);
+
+        // Set domain target address to user2_addr, this should trigger auto migration
+        router::set_target_addr(user, domain_name, option::none(), user2_addr);
+        {
+            let v1_target_address = router::get_v1_target_addr(domain_name, option::none());
+            assert!(option::is_none(&v1_target_address), 1);
+            let v2_target_address = router::get_v2_target_addr(domain_name, option::none());
+            assert!(*option::borrow(&v2_target_address) == user2_addr, 2);
+        };
+        // Set subdomain target address to user2_addr
+        // This should throw ENOT_OWNER_OF_NAME error because we do not auto migrate subdomain and set target address for v1 name in MODE_V1_AND_V2 is not allowed
```

Comment:
ATTENTION: i really think throwing `ENOT_OWNER_OF_NAME` would be very confusing to user, as user can't tell it's actually they need to migrate to fix it.